.babelrc.js

```diff
@@ -1,48 +1,67 @@
 "use strict";
 
-const env = process.env.BABEL_ENV || process.env.NODE_ENV;
-const envOpts = {
-  loose: true,
-};
+module.exports = function(api) {
+  const env = api.env();
 
-switch (env) {
-  case "development":
-    envOpts.debug = true;
-  // fall-through
-  case "test":
-  case "cov":
-    envOpts.targets = {
-      node: "current",
-    };
-}
+  const envOpts = {
+    loose: true,
+    modules: false,
+    exclude: ["transform-typeof-symbol"],
+  };
 
-const config = {
-  comments: false,
-  presets: [["@babel/env", envOpts]],
-  plugins: [
-    // TODO: Use @babel/preset-flow when
-    // https://github.com/babel/babel/issues/7233 is fixed
-    "@babel/plugin-transform-flow-strip-types",
-    ["@babel/proposal-class-properties", { loose: true }],
-    "@babel/proposal-export-namespace-from",
-    "@babel/proposal-numeric-separator",
-    ["@babel/proposal-object-rest-spread", { useBuiltIns: true }],
-  ],
-  overrides: [
-    {
-      test: "packages/babylon",
-      plugins: [
-        "babel-plugin-transform-charcodes",
-        ["@babel/transform-for-of", { assumeArray: true }],
-      ],
-    },
-  ],
-};
+  let convertESM = true;
+
+  switch (env) {
+    // Configs used during bundling builds.
+    case "babylon":
+    case "standalone":
+      convertESM = false;
+      break;
+    case "production":
+      // Config during builds before publish.
+      break;
+    case "development":
+      envOpts.debug = true;
+      envOpts.targets = {
+        node: "current",
+      };
+      break;
+    case "test":
+    case "cov":
+      envOpts.targets = {
+        node: "current",
+      };
+      break;
+  }
 
-// we need to do this as long as we do not test everything from source
-if (env === "cov") {
-  config.auxiliaryCommentBefore = "istanbul ignore next";
-  config.plugins.push("babel-plugin-istanbul");
-}
+  const config = {
+    comments: false,
+    presets: [["@babel/env", envOpts]],
+    plugins: [
+      // TODO: Use @babel/preset-flow when
+      // https://github.com/babel/babel/issues/7233 is fixed
+      "@babel/plugin-transform-flow-strip-types",
+      ["@babel/proposal-class-properties", { loose: true }],
+      "@babel/proposal-export-namespace-from",
+      "@babel/proposal-numeric-separator",
+      ["@babel/proposal-object-rest-spread", { useBuiltIns: true }],
+    ].concat(convertESM ? ["@babel/transform-modules-commonjs"] : []),
```

Comment:
not that different but `convertESM ? ["@babel/transform-modules-commonjs"] : []` and `.filter(Boolean)` works too

Reply:
Great call, I'll do that instead.